agent/api/task/task_attachment_handler.go

```diff
@@ -25,7 +25,7 @@ import (
 // AttachmentHandler defines an interface to handel attachment received from ACS.
 type AttachmentHandler interface {
 	parseAttachment(acsAttachment *ecsacs.Attachment) error
-	validateAttachment(acsTask *ecsacs.Task) error
+	validateAttachment(acsTask *ecsacs.Task, task *Task) error
```

Comment:
nit: Instead of passing in both type of tasks, it might be enough to pass the Network mode.

Reply:
I was thinking that future attachment might want to take other fields from the internal task struct.